docs/tutorials/migrating-your-ga4-data-to-snowplow/index.md

```diff
@@ -0,0 +1,204 @@
+To create your service account using the cli, follow the steps below:
+
+##### Install gcloud
+See https://cloud.google.com/sdk/docs/install-sdk for installation instructions
+
+##### Authorize gcloud
+```shell
+gcloud auth login
+```
+
+##### Set the following variables in your terminal
+- `PROJECT_ID`: the name of the GCP project in which the GA4 data is stored
+- `SERVICE_ACCOUNT_NAME`: The name to give the new service account
+- `ROLE_NAME`: The name to give the new custom role
+- `KEY_FILE_NAME`: The name to give the the service account key file that is downloaded. Use the .json extension
+e.g.
+```shell
+export PROJECT_ID=project_123_abc
+export SERVICE_ACCOUNT_NAME=migrator-service-account
+export ROLE_NAME=ga4_exporter_role
+export KEY_FILE_NAME=migrator_service_account_key.json
+```
+
+##### Create a service account 
+```shell
+gcloud iam service-accounts create $SERVICE_ACCOUNT_NAME --display-name $SERVICE_ACCOUNT_NAME
+```
+*Note: this will error if the service account name uses underscores, or if the service account already exists*
+
+##### Create a custom role
+```shell
+gcloud iam roles create $ROLE_NAME --project=$PROJECT_ID --permissions=bigquery.tables.export,bigquery.tables.getData,bigquery.tables.get,bigquery.tables.create,bigquery.tables.update,bigquery.tables.updateData,bigquery.tables.list,bigquery.tables.delete,bigquery.jobs.create,storage.objects.create,storage.objects.get,storage.objects.list,storage.objects.delete,storage.buckets.getIamPolicy,storage.buckets.setIamPolicy,storage.buckets.create,storage.buckets.get,bigquery.datasets.create,bigquery.routines.create,bigquery.routines.get
+```
+*Note this will error if you already have a custom role with this name*
+
+
+##### Grant the custom role to the service account
+```shell
+gcloud projects add-iam-policy-binding $PROJECT_ID --member serviceAccount:$SERVICE_ACCOUNT_NAME@$PROJECT_ID.iam.gserviceaccount.com --role projects/$PROJECT_ID/roles/$ROLE_NAME
+```
+
+##### Create a key for the service account
+```shell
+gcloud iam service-accounts keys create $KEY_FILE_NAME --iam-account $SERVICE_ACCOUNT_NAME@$PROJECT_ID.iam.gserviceaccount.com
+```
+*This will download the key to your local machine in the same folder in which you are running the command*
+</TabItem>
+
+<TabItem value="Terraform" label="Terraform">
+##### Install gcloud
+See https://cloud.google.com/sdk/docs/install-sdk for installation instructions
+
+##### Authorize gcloud
+```shell
+gcloud auth application-default login
+```
+
+##### Install Terraform
+See https://learn.hashicorp.com/tutorials/terraform/install-cli for installation instructions
+
+##### Run Terraform
+There is a file in the root directory of the migrator tool called `manage_service_account.tf`. This file contains the Terraform code to create the service account.
+To create the service account in the root directory of the migrator tool, run the following commands:
+```shell
+terraform init
+terraform apply
```

Comment:
ah ok, yeah i made a couple of changes to the project id to add it as a variable in the same way role id and SA id are added, so that the user can put all three in a tf vars file. 
It errored when i first tried to run your tf code, and i didn't realise until after that in the Issue you suggested to run it as `GOOGLE_PROJECT="myprojectid" terraform apply` if that's better than my changes, happy to change it back

Reply:
Oh, nope that change is fine I just didn't realise it happened. :upside_down_face: All good for this, then!